src/main/java/com/opentable/db/postgres/embedded/PreparedDbProvider.java

```diff
@@ -196,25 +215,46 @@ private static void create(final DataSource connectDb, final String dbName, fina
         }
     }
 
-    private static class DbInfo
+    public static class DbInfo
     {
+        public static DbInfo ok(final String dbName, final int port, final String user) {
```

Comment:
Incidentally this would be a great place to have an [`Either<E, R>`](https://docs.atlassian.com/fugue/2.4.0/fugue/apidocs/com/atlassian/fugue/Either.html) type available. :)

Reply:
Yes, but in the absence of that I tend (with bounded results) to make constructors private and use static helpers.

Reply:
Right but without it `DbInfo` exposes a really weird public interface purely as a side effect of it having 2 semantic states (a valid connection info, or an error). Because we're opening this class up to the world, end-users have to be aware of the mutually incompatible ways this class can be initialized and act appropriately (eg, by ensuring it doesn't have an exception). Most probably for your actual use case what we want to expose is a semantically simpler `ConnectionInfo` class that is always fully initialized, allowing `DbInfo` as this dual-state type to remain an internal implementation detail.